File: source_data/figure_code/Fig4_hardfilter.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""Figure 4 — GATK INDEL hard-filtering before/after (GATK4, full depth). Reads source_data.xlsx."""
import pandas as pd, numpy as np, matplotlib.pyplot as plt

d = pd.read_excel('../source_data/source_data.xlsx', sheet_name='Fig4_hardfilter')
types = ['SNV', 'INDEL']; filts = [('raw (unfiltered)', '#B0B0B0'), ('hard-filtered', '#4C72B0')]
x = np.arange(len(types)); w = 0.36
fig, axes = plt.subplots(1, 2, figsize=(10.5, 4.6))
for ax, met in zip(axes, ['F1', 'precision']):
    for i, (fv, col) in enumerate(filts):
        vals = [d[(d.variant_type == vt) & (d['filter'] == fv)][met].values for vt in types]
        vals = [v[0] if len(v) else np.nan for v in vals]
        b = ax.bar(x + (i - .5) * w, vals, w, color=col, label=fv)
        ax.bar_label(b, fmt='%.3f', fontsize=8)
    ax.set_xticks(x); ax.set_xticklabels(types); ax.set_title(met); ax.set_ylim(.6, 1.03); ax.grid(alpha=.3, axis='y')
axes[0].legend(fontsize=9, loc='lower left')
fig.suptitle('Figure 4 — GATK INDEL hard-filtering before/after (full depth)')
fig.tight_layout(); fig.savefig('Fig4_hardfilter.png', dpi=300); print('-> Fig4_hardfilter.png')

File: source_data/figure_code/Fig4_hardfilter.R
# Figure 4 — GATK INDEL hard-filtering before/after (GATK4, full depth). Reads source_data.xlsx.
library(readxl); library(ggplot2); library(dplyr); library(tidyr)
d <- read_excel("../source_data/source_data.xlsx", sheet = "Fig4_hardfilter") |>
  pivot_longer(c(F1, precision), names_to = "metric", values_to = "value")
ggplot(d, aes(variant_type, value, fill = filter)) +
  geom_col(position = position_dodge(.7), width = .6) +
  geom_text(aes(label = sprintf("%.3f", value)), position = position_dodge(.7), vjust = -.3, size = 3) +
  facet_wrap(~metric) + coord_cartesian(ylim = c(.6, 1.03)) +
  scale_fill_manual(values = c(`raw (unfiltered)` = "#B0B0B0", `hard-filtered` = "#4C72B0")) +
  labs(title = "Figure 4 — GATK INDEL hard-filtering before/after (full depth)", x = NULL, y = NULL) +
  theme_bw() + theme(legend.position = "bottom")
ggsave("Fig4_hardfilter_R.png", width = 10.5, height = 4.8, dpi = 300)
